numpy/core/src/multiarray/refcount.c.src

```diff
@@ -350,3 +323,38 @@ _fillobject(char *optr, PyObject *obj, PyArray_Descr *dtype)
     }
     return;
 }
+
+
+/*NUMPY_API
+ * Assumes contiguous
+ */
+NPY_NO_EXPORT void
+PyArray_FillObjectArray(PyArrayObject *arr, PyObject *obj)
+{
+    npy_intp i,n;
+    n = PyArray_SIZE(arr);
+    if (PyArray_DESCR(arr)->type_num == NPY_OBJECT) {
+        PyObject **optr;
+        optr = (PyObject **)(PyArray_DATA(arr));
+        n = PyArray_SIZE(arr);
+        if (obj == NULL) {
+            for (i = 0; i < n; i++) {
+                *optr++ = NULL;
```

Comment:
The code path `obj == NULL` is never hit

Reply:
Yes, but it is exposed API, note that this function was not touched, the file was just modified so much it appears to be changed. Also it could just as well use `XINCREF` and remove the branching, I doubt it is worth to micro-optimize...